devimint/src/util.rs

```diff
@@ -1071,6 +1073,52 @@ pub fn is_backwards_compatibility_test() -> bool {
     is_env_var_set(FM_BACKWARDS_COMPATIBILITY_TEST_ENV)
 }
 
+/// Sets the fedimint-cli binary to match the fedimintd's version, which is
+/// needed for running DKG. Returns the original fedimint-cli path and mint
+/// client alias so the caller can reset the fedimint-cli version after DKG
+pub async fn use_matching_fedimint_cli_for_dkg() -> Result<(String, String)> {
+    let pkg_version = semver::Version::parse(env!("CARGO_PKG_VERSION"))?;
+    let fedimintd_version = crate::util::FedimintdCmd::version_or_default().await;
+    let original_fedimint_cli_path = crate::util::get_fedimint_cli_path().join(" ");
+
+    if pkg_version == fedimintd_version {
+        // we're on the current version if the fedimintd version is the same as the
+        // package version. to use the current version of `fedimint-cli` built by cargo,
+        // we need to unset FM_FEDIMINT_CLI_BASE_EXECUTABLE
+        std::env::remove_var(FM_FEDIMINT_CLI_BASE_EXECUTABLE_ENV);
+    } else {
+        let parsed_fedimintd_version = fedimintd_version.to_string().replace(['-', '.'], "_");
+
+        // matches format defined by nix_binary_version_var_name in scripts/_common.sh
+        let fedimint_cli_path_var = format!("fm_bin_fedimint_cli_v{parsed_fedimintd_version}");
+        let fedimint_cli_path = std::env::var(fedimint_cli_path_var)?;
+        std::env::set_var(FM_FEDIMINT_CLI_BASE_EXECUTABLE_ENV, fedimint_cli_path);
```

Comment:
BTW. For everyone's awarness: technically `set_var` is not thread-safe and it's probably best to architect programs to avoid changing their own environment while running: https://doc.rust-lang.org/std/env/fn.set_var.html#safety

This is distinct from setting environment of your children programs when spawning them, which is OK.

Not a blocker for this PR, not an immediate issue, but something to be aware going forward.

Ping @maan2003